drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c

```diff
@@ -267,14 +273,17 @@ static int buffer_init(struct vb2_buffer *vb)
 {
 	struct bm2835_mmal_dev *dev = vb2_get_drv_priv(vb->vb2_queue);
 	struct vb2_v4l2_buffer *vb2 = to_vb2_v4l2_buffer(vb);
-	struct mmal_buffer *buf = container_of(vb2, struct mmal_buffer, vb);
+	struct vb2_mmal_buffer *buf =
+				container_of(vb2, struct vb2_mmal_buffer, vb);
 
 	v4l2_dbg(1, bcm2835_v4l2_debug, &dev->v4l2_dev, "%s: dev:%p, vb %p\n",
 		 __func__, dev, vb);
-	buf->buffer = vb2_plane_vaddr(&buf->vb.vb2_buf, 0);
-	buf->buffer_size = vb2_plane_size(&buf->vb.vb2_buf, 0);
+	buf->mmal.buffer = vb2_plane_vaddr(&buf->vb.vb2_buf, 0);
+	buf->mmal.buffer_size = vb2_plane_size(&buf->vb.vb2_buf, 0);
+
+	mmal_vchi_buffer_init(dev->instance, &buf->mmal);
```

Comment:
Is this init function guaranteed never to fail? You don't mention this change,

Reply:
Github loses the commit the reviewer was looking at when they made the comment, so it's possible this has become misplaced in the series.
A later patch does check the return value, but this one ought to too.

The vchi context is now attached to the buffer at allocation, not allocated on demand. I'll review to check that bits got added at the correct point.